src/tools/sc-hsm-tool.c

```diff
@@ -157,7 +160,11 @@ static int generatePrime(BIGNUM *prime, const BIGNUM *s, const int bits, unsigne
 
 	do {
 		// Generate random prime
+#if OPENSSL_VERSION_NUMBER  >= 0x00908000L /* last parm is BN_GENCB which is null in our case */
+		BN_generate_prime_ex(prime, bits, 1, NULL, NULL, NULL);
```

Comment:
It could be done, but this is the only place in OpenSC it is used. 
I would like @CardContact to look at all the changes in this module first.